File: aws_lambda_functions/fetchTeeTimesPY/course.py
from dataclasses import dataclass
from typing import Dict, Any


@dataclass
class Course:
    id: int
    name: str
    display_name: str
    club_name: str
    city: str
    rating: float
    external_api: str
    external_api_attributes: Dict[str, Any]
    booking_visibility_days: int = 0

    @classmethod
    def from_dict(cls, data: Dict[str, Any]) -> 'Course':
        # Handle nested city data from join
        city_name = data.get('cities', {}).get('name') if 'cities' in data else data.get('city', '')
        
        return cls(
            id=data['id'],
            name=data['name'],
            display_name=data['display_name'],
            club_name=data['club_name'],
            city=city_name,
            rating=data['rating'],
            external_api=data['external_api'],
            external_api_attributes=data.get('external_api_attributes', {}),
            booking_visibility_days=data.get('booking_visibility_days', 0)
        )

    def to_dict(self) -> Dict[str, Any]:
        return {
            'id': self.id,
            'name': self.name,
            'display_name': self.display_name,
            'club_name': self.club_name,
            'city': self.city,
            'rating': self.rating,
            'external_api': self.external_api,
            'external_api_attributes': self.external_api_attributes,
            'booking_visibility_days': self.booking_visibility_days
        } 

File: aws_lambda_functions/fetchTeeTimes-CHRONO-puppeteer/Course.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Course = void 0;
class Course {
    constructor(id, name, display_name, club_name, city, rating, external_api, external_api_attributes = {}, booking_visibility_days = 0, booking_visibility_start_time, timezone) {
        this.id = id;
        this.name = name;
        this.display_name = display_name;
        this.club_name = club_name;
        this.city = city;
        this.rating = rating;
        this.external_api = external_api;
        this.external_api_attributes = external_api_attributes;
        this.booking_visibility_days = booking_visibility_days;
        this.booking_visibility_start_time = booking_visibility_start_time;
        this.timezone = timezone;
    }
    // Helper method to create a Course from a plain object
    static fromObject(obj) {
        // Handle nested city data from join
        const cityName = obj.cities?.name;
        return new Course(obj.id, obj.name, obj.display_name, obj.club_name, cityName, obj.rating, obj.external_api, obj.external_api_attributes || {}, obj.booking_visibility_days || 0, obj.booking_visibility_start_time, obj.timezone);
    }
    // Helper method to convert Course to a plain object
    toObject() {
        return {
            id: this.id,
            name: this.name,
            display_name: this.display_name,
            club_name: this.club_name,
            city: this.city,
            rating: this.rating,
            external_api: this.external_api,
            external_api_attributes: this.external_api_attributes,
            booking_visibility_days: this.booking_visibility_days,
            booking_visibility_start_time: this.booking_visibility_start_time,
            timezone: this.timezone
        };
    }
}
exports.Course = Course;
